server/s3_copy_object_action.cc

```diff
@@ -221,6 +221,43 @@ void S3CopyObjectAction::fetch_source_object_info_failed() {
 // Validate source bucket and object
 void S3CopyObjectAction::validate_copyobject_request() {
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  auto meta_directive_hdr =
+      request->get_header_value("x-amz-metadata-directive");
+  auto tagging_directive_hdr =
+      request->get_header_value("x-amz-tagging-directive");
+
+  if (!meta_directive_hdr.empty()) {
+    s3_log(S3_LOG_DEBUG, stripped_request_id,
+           "Received x-amz-metadata-directive header, value: %s",
+           meta_directive_hdr.c_str());
+    if (meta_directive_hdr == DirectiveValueReplace) {
+      s3_put_action_state = S3PutObjectActionState::validationFailed;
+      set_s3_error("NotImplemented");
+      send_response_to_s3_client();
+      return;
+    } else if (meta_directive_hdr != DirectiveValueCOPY) {
```

Comment:
HTTP standard states case insensitivity

Reply:
AWS is mandating the case-sensitivity. I tested it with AWS. Included the response in other comment.